grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -520,6 +523,58 @@ class ProductAvailabilityService {
         return quantityMap
     }
 
+    /**
+     * Find product availability record grouped by product that satisfy the filters which contain primarily:
+     * expiration (we can include/exclude inventory items that expire in 30/90/180/365 days),
+     * additionalLocations - locations to search (and sum) the stock in (other than current location)
+     */
+    List<InventoryByProduct> getInventoriesByProduct(ReorderReportFilterCommand command) {
+        List<Location> locations = [AuthService.currentLocation]
+        if (command.additionalLocations) {
+            locations.addAll(command.additionalLocations)
+        }
+        return ProductAvailability.createCriteria().list {
+            projections {
+                sum("quantityOnHand")
+                sum("quantityAvailableToPromise")
+                groupProperty("product")
+            }
+            inList("location", locations)
```

Comment:
why `inList` and not `eq`? You only have a single location

Reply:
We have at least one location (current location), but you can pass `additionalLocations` so that you end up with the list.
I thought it would be form oversubstance if I were to do :
```groovy
if (command.additionalLocations) {
  inList("location", locations)
} else {
  eq("location", AuthService.currentLocation
}
```
What do you think?